### PR TITLE
DX-1019: Read Your Writes

### PR DESCRIPTION
**Pull request by fahreddinozcan**

### CODE REVIEW

File: upstash_redis/client.py [hunk 1]
```diff
@@ -44,6 +45,7 @@ def __init__(
         :param rest_retries: how many times an HTTP request will be retried if it fails
         :param rest_retry_interval: how many seconds will be waited between each retry
         :param allow_telemetry: whether anonymous telemetry can be collected
+        :param read_your_writes: whether the client should wait for the response of a write operation before sending the next one
```
Comment (mdogan) [inline upstash_redis/client.py:48]: Not exactly a wait... Writing the same comment from JS SDK PR:
```
When this flag is enabled, any subsequent commands issued by this client are guaranteed to observe the effects of all earlier writes submitted by the same client.
```

**Comment by mdumandag** [comment]
I have fixed the problems in the tests, added asyncio support. Only the pipeline tests are not passing because the fix for them are not deployed to instance we are running the test on. If you want, I can disable these tests and merge this PR as it is

File: upstash_redis/client.py (hunk 1 above)
Comment (mdumandag) [reply upstash_redis/client.py]: Updated the docstring

**Comment by fahreddinozcan** [comment]
Cloud team stated that all replicas are updated to the most recent version. I guess this shouldn't be happening?

**Comment by mdumandag** [comment]
Oh, I assumed that since pipeline tests were failing 1.10.11 might not be deployed yet, but it was. I think we also need to change some logic in pipeline implementation to make it work. Let me send a fix for it

**Pull request merged**